### DISCUSSION TITLE
Allowing password override when no user specified in [databases] with auth type PAM

### DISCUSSION THREAD

**Discussion by floptical** [pgbouncer/pgbouncer#856] [Q&A]
Hi all, I have a pressing need for a very unique feature here and I am brand new to C, hoping someone can point me to the right spot in the code that I can try modifying and then compiling myself.

So I noticed that when you have a backend connection like this, Pgbouncer will proceed to override the client-provided username and password I pass to my PAM module and use the connection string for the backend connection:

`adb = user=some_user password=some_password host=dbhost.com dbname=somedb port=5432`

 If I state only a user, it will use that user (per the documentation) and the client password:

`adb = user=some_user host=dbhost.com dbname=somedb port=5432`

However, if I only have a password listed, it is not overwritten and instead the client password is used:

`adb = password=some_password host=dbhost.com dbname=somedb port=5432`

 I am hoping to change this behavior but am having trouble understanding the code base. I see a "forced_user" property under the "db" structure (from the pgDatabase static variable), but I can't find where that's actually set/read from the ini config file.
 
 Any assistance is greatly appreciated.

**Comment by petere** [comment]
I think this is probably kind of a bug. The password specification should be rejected if no user is specified along with it.

However, I'm not sure whether that is what you are trying to fix?
